.github/workflows/integration-build-product.yml

```diff
@@ -529,7 +584,21 @@ jobs:
           docker pull "${{ env.MID_REGISTRY }}/ci/${{ inputs.product_name }}-${{ matrix.distro }}:${{ needs.Build-Spawner.outputs.raised_version }}"
           docker tag "${{ env.MID_REGISTRY }}/ci/${{ inputs.product_name }}-${{ matrix.distro }}:${{ needs.Build-Spawner.outputs.raised_version }}" "${{ env.REGISTRY }}/qa/${{ inputs.product_name }}-${{ matrix.distro }}:${{ needs.Build-Spawner.outputs.raised_version }}"
           docker tag "${{ env.MID_REGISTRY }}/ci/${{ inputs.product_name }}-${{ matrix.distro }}:${{ needs.Build-Spawner.outputs.raised_version }}" "${{ env.PUSH_REGISTRY }}/qa/${{ inputs.product_name }}-${{ matrix.distro }}:${{ needs.Build-Spawner.outputs.raised_version }}"
-
+      
+      - name: Update robot jsons to use qa project
+        if: ${{ github.event_name == 'pull_request' }}
+        shell: bash
+        run: |
+          # iterate over .json files and update spawner.image to use ci image
```

Comment:
repeated code here...
can't we prepare, store and reuse the jsons instead of creating them twice ?

Reply:
maybe in `Validate-boostrap-configs` step ?